extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/device/EDS006x.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal.device;
+
+import static org.eclipse.smarthome.binding.onewire.internal.OwBindingConstants.*;
+
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.Illuminance;
+import javax.measure.quantity.Pressure;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.binding.onewire.internal.OwException;
+import org.eclipse.smarthome.binding.onewire.internal.Util;
+import org.eclipse.smarthome.binding.onewire.internal.handler.OwBaseBridgeHandler;
+import org.eclipse.smarthome.binding.onewire.internal.handler.OwBaseThingHandler;
+import org.eclipse.smarthome.binding.onewire.internal.owserver.OwserverDeviceParameter;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.MetricPrefix;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EDS006x} class defines an EDS006x device
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class EDS006x extends AbstractOwDevice {
+    private final Logger logger = LoggerFactory.getLogger(EDS006x.class);
+
+    private final OwDeviceParameterMap temperatureParameter = new OwDeviceParameterMap() {
+        {
+            set(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/temperature"));
```

Comment:
Wouldn't it be much more readable to add another `OwDeviceParameterMap` constructor that takes a thing type and a `OwserverDeviceParameter` as an argument?

Reply:
I'm still think of other bridge types which would require other device parameters (e.g. a bridge that directly uses the DS9490 USB Onewire-Master would need the register inside the device instead of a path). This would require something like

```
private final OwDeviceParameterMap temperatureParameter = new OwDeviceParameterMap() {
    {
        set(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/temperature"));
        set(THING_TYPE_DS9490R, new Ds9490RDeviceParameter(0x4711));
    }
```

I don't think that can be done with constructors.

Reply:
If we agree that this will never happen, the code could be refactored and simplified. But that is quite large (not sure if we would need a CQ for that).

Reply:
Or we add a constructor:

```Java
public OwDeviceParameterMap(final Map<ThingTypeUID, OwDeviceParameter> map) {
    this.map.putAll(map);
}
```

Then you could add multiple key value pairs on construction or a single one.

```Java
private final OwDeviceParameterMap temperatureParameter = new OwDeviceParameterMap(Collections.singletonMap(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/temperature")));
```

I don't understand the current implementation of the `set` method of `OwDeviceParameterMap`.
There is a check if the key is already contained to call `replace` or `put`.
Why not calling `put` without additional checks?

Reply:
Is that really more readable for let's say three parameter sets? Looking at [this discussion](https://stackoverflow.com/questions/8261075/adding-multiple-entries-to-a-hashmap-at-once-in-one-statement) it seems that double-brace-method is most clear if we stay away from Guava and don't have Java 9 at hand. 

Another argument not to change it now: This is the way it is for all other devices. As I said before: we could think of removing the map totally and giving up the idea of other bridges. With the coming changes in #6764 it 's becoming more and more difficult and would require a rewrite of large parts of the OWFS code to get the same functionality. But IMO this is out of scope of this PR.

Regarding the latter remark: I don't remember. Probably it wasn't a HashMap in a previous version during development. I have removed the check.

Reply:
@J-N-K You can leave it as it is if you want.

But AFAIK using that code you create for every variable a new inner anonymous class. So instead of using the `OwDeviceParameterMap` class and create a new instance of it, you create a new class and a new instance of that anonymous class for every variable (there should be class files with a '$' for every instance in your jar).
I personally see no reason for new classes as it can be handled all using the `OwDeviceParameterMap` class (that only wraps a map).

Reply:
I would leave it as it is now and keep it on my not-so-high priority list to refactor that for all devices in the future.